File: backend/src/main/java/tom/projects/monkeylog/common/dto/Page.java
package tom.projects.monkeylog.common.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.domain.Slice;

import java.util.List;

@Data
@AllArgsConstructor
public class Page<T> {
    @NotNull
    private Integer number;
    @NotNull
    private Integer size;
    @NotNull
    private Integer numberOfElements;
    @NotNull
    private List<T> content;

}

File: backend/src/main/kotlin/tom/com/workout/logbook/common/dto/Page.kt
package tom.com.workout.logbook.common.dto

data class Page<T>(
    val isFirst: Boolean,
    val isLast: Boolean,
    val totalPages: Int,
    val totalElements: Long,
    val number: Int,
    val size: Int,
    val content: List<T>,
)
